frontend/src/actions/events.js

```diff
@@ -1,39 +1,44 @@
-import { genericApi } from "../api/genericApi";
+import {genericApi} from "../api/genericApi";
+import {showSnackbar} from "./snackbar";
+import {SUCCESS} from "../components/Snackbar/SnackbarSeverityConstants";
 
-export const viewEventDetails = (event) => {
-  return {
-    type: "VIEW_EVENT_DETAILS",
-    value: event,
-  };
-};
-
-export const viewUpcomingEventsOnly = (events) => {
-  return {
-    type: "VIEW_UPCOMING_ONLY",
-    events: events,
-  };
-};
+export const getEventPageData = (eventId) => async (dispatch) => {
+  dispatch(getEvent(eventId));
+  dispatch(getEventParticipants(eventId));
+}
 
-export const viewAllEvents = (events) => {
-  return {
-    type: "VIEW_ALL_EVENTS",
-    events: events,
-  };
-};
-
-export const participantJoin = async (dispatch, eventId, userId, username) => {
+export const getEventParticipants = (eventId) => async dispatch => {
   try {
-    const res = await genericApi.patch(`/api/events/${eventId}/participants`, {
-      participant: {
-        uid: userId,
-        displayName: username,
-      },
-    });
+    const res = await genericApi.get(`/api/events/${eventId}/participants`);
     dispatch({
-      type: "PARTICIPANT_JOIN",
-      payload: res.data,
+      type: "GET_EVENT_PARTICIPANTS",
+      payload: res.data
     });
-    getEvent(dispatch, eventId);
+  } catch (e) {
+    dispatch({
+      type: "ERROR_EVENT_PARTICIPANTS",
+      payload: e.message
+    })
+  }
+}
+
+export const participantJoin = async (dispatch, eventId, userId) => {
+  try {
+    const res = genericApi.patch(`/api/events/${eventId}/participants`, {
+      userId: userId,
+    })
```

Comment:
Do you need all these dispatches in order? And does dispatch return a promise?
I usually just await the api request and then dispatch the rest without being in .then()'s.

Reply:
![image](https://user-images.githubusercontent.com/42756146/128782049-101fa90b-b2c9-4ea0-8a37-dc7761b51c8b.png)
I followed you, actually, since I was basing the work off of `addMember`